src/request.rs

```diff
@@ -15,64 +15,116 @@ pub(crate) enum Method<T: Serialize> {
 #[cfg(not(target_arch = "wasm32"))]
 pub(crate) async fn request<Input: Serialize, Output: DeserializeOwned + 'static>(
     url: &str,
-    apikey: &str,
+    apikey: Option<&str>,
     method: Method<Input>,
     expected_status_code: u16,
 ) -> Result<Output, Error> {
     use isahc::http::header;
     use isahc::*;
 
-    let auth = format!("Bearer {}", apikey);
     let user_agent = qualified_version();
 
     let mut response = match &method {
         Method::Get => {
-            Request::get(url)
-                .header(header::AUTHORIZATION, auth)
-                .header(header::USER_AGENT, user_agent)
-                .body(())
-                .map_err(|_| crate::errors::Error::InvalidRequest)?
-                .send_async()
-                .await?
+            if let Some(key) = apikey {
+                let auth = format!("Bearer {}", key);
+                Request::get(url)
+                    .header(header::AUTHORIZATION, auth)
+                    .header(header::USER_AGENT, user_agent)
+                    .body(())
+                    .map_err(|_| crate::errors::Error::InvalidRequest)?
+                    .send_async()
+                    .await?
+            } else {
+                Request::get(url)
+                    .header(header::USER_AGENT, user_agent)
+                    .body(())
+                    .map_err(|_| crate::errors::Error::InvalidRequest)?
+                    .send_async()
+                    .await?
+            }
```

Comment:
```suggestion
            let mut request = Request::get(url)
              .header(header::USER_AGENT, user_agent);
            
            if let Some(key) = apikey {
                let auth = format!("Bearer {}", key);
                request = request.header(header::AUTHORIZATION, auth);
            }
            request
                    .body(())
                    .map_err(|_| crate::errors::Error::InvalidRequest)?
                    .send_async()
                    .await?
            }
```

Can you do the same for all the verbs, please